controllers/synapse/synapse_mautrixsignal_serviceaccount.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package synapse
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	synapsev1alpha1 "github.com/opdev/synapse-operator/apis/synapse/v1alpha1"
+)
+
+// serviceAccountForMautrixSignal returns a ServiceAccount object for running the mautrix-signal bridge
+func (r *SynapseReconciler) serviceAccountForMautrixSignal(s *synapsev1alpha1.Synapse, objectMeta metav1.ObjectMeta) (client.Object, error) {
+	sa := &corev1.ServiceAccount{
+		ObjectMeta: objectMeta,
+	}
+
+	// Set Synapse instance as the owner and controller
+	if err := ctrl.SetControllerReference(s, sa, r.Scheme); err != nil {
+		return &corev1.ServiceAccount{}, err
+	}
+	return sa, nil
+}
+
+// roleBindingForMautrixSignal returns a RoleBinding object for the mautrix-signal bridge
+func (r *SynapseReconciler) roleBindingForMautrixSignal(s *synapsev1alpha1.Synapse, objectMeta metav1.ObjectMeta) (client.Object, error) {
+	rb := &rbacv1.RoleBinding{
+		ObjectMeta: objectMeta,
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     "system:openshift:scc:anyuid",
```

Comment:
I guess this answers some of my questions. I'm assuming some pods are coming with root powered containers. Nothing against it for now but we should point some backlog issues for the future to restrain permissions before the whole thing becomes too hard to change.

Reply:
Let's discuss this in face to face

Reply:
Sure we can put this in our agenda for the next meeting. Again no worries with that for the moment.